RHIDP-7542: Template Editor section update (#1178)
redhat-developer/red-hat-developers-documentation-rhdh: modules/customizing-templates/proc-creating-templates.adoc

:_mod-docs-content-type: PROCEDURE
[id="proc-creating-templates_{context}"]

> = Creating a Software Template by using the Template Editor templates

Reviewer: The word "template" here seems out of place, no?

You can create a Software Template by using the Template Editor templates.

> Alternately, you can use the Template Editor to do any of the following actions:
> 
> * *File > Open template directory*
> * *File > Create template directory*
> * *File > Close template editor*
> * *Custom Fields Explorer*
> * *Installed Actions Documentation*

Reviewer 2: I'm sorry, but it doesn't seem like all of the following items in this list are _actions_. I think of action as something like _click_ or _select_ or _navigate_.  Open, Create, Close make sense (perhaps without the _File >_ part) but Custom Fields Explorer and Installed Actions Documentation make the list unparallel. Maybe try something more like this? 

```suggestion
Alternately, you can use the Template Editor to do any of the following actions:

* *Open template directory*
* *Create template directory*
* *Close template editor*
* Use the *Custom Fields Explorer* to [perform action]. 
* View *Installed Actions Documentation*
```
I don't know if _View_ is the right action for the Installed Actions Documentation, so you can replace that verb if needed

> * Use the *Custom Fields Explorer* button to test custom fields in your 'templates.yaml' file

Reviewer 2: ```suggestion
* Use the *Custom Fields Explorer* button to test custom fields in your `templates.yaml` file
```

.Procedure

To create a Software Template by using the Template Editor templates, complete the following steps:

. In your {product} navigation menu, click *Catalog > Self-service*.
. Click the overview menu and select *Manage Templates*.
+
image::rhdh/template-editor.png[Template Editor]
+

[NOTE]
====
* The following options on the *Managed Templates* page do not create a software component in your {product} instance:
** *Template Form Playground*: Use to create and test the `templates.yaml` file
** *Custom Field Explorer*: Use to test custom fields
====

. On the *Managed Templates* page, select any of the following options:
* *Load Template Directory*: Use to load an existing `templates.yaml` file
** In your local file manager, navigate to the folder where your `templates.yaml` file is stored and click *Select*.
* *Create New Template*: Use to create a `templates.yaml` file
** In your local file manager, navigate to the folder where you want the Template Editor to create a `templates.yaml` file and click *Select*.
.. On the *Template Editor* page, select the `templates.yaml` file.

> .. On the *Fill in some steps* tab, enter text into the required fields and click *Next*.

Reviewer: IMO, this doesn't make sense as an instruction. It could be added as a note perhaps, like "you can use the preview on the right to test your template"

.. On the *Repository Location* tab, enter text into the required fields and click *Review*.
.. Optional: Modify the YAML definition for the parameters of your template. For more information about these parameters, see <<Creating a Software Template as a YAML file>>.

> .. Review the information for accuracy, then click *Create*.

Reviewer 2: Oops, looks like all of this got bolded by mistake

> .. Optional: Modify the YAML definition for the parameters of your template. For more information about these parameters, see <<Creating a Software Template as a YAML file>>.

Reviewer: I think the next action needs to be `Importing an existing template to Red Hat Developer Hub`. Now that I have a template.yaml.

:_mod-docs-content-type: PROCEDURE
[id="proc-creating-templates_{context}"]

> = Creating a Software Template by using the Template Editor Templates

Reviewer 3: ```suggestion
= Creating a Software Template by using the Template Editor templates
```

You can create a Software Template by using the Template Editor templates.

.Procedure

. In your {product} navigation menu, click *Catalog > Self-service*.
. Click the overview menu button and select *Manage Templates*.
+
image::rhdh/template-editor.png[Template Editor]
+

[NOTE]
====
* The following options on the *Managed Templates* page do not create a software component in your {product} instance:

> ** *Template Form Playground*: Use to create and test the templates.yaml file

Reviewer 2: ```suggestion
** *Template Form Playground*: Use to create and test the `templates.yaml` file
```

** *Custom Field Explorer*: Use to test custom fields
====

> . On the *Managed Templates* page, select any of the following options:

Reviewer 3: Would the steps differ depending on which option the user selects from the following list? If yes, then that should be called out.

** *Load Template Directory*
** *Create New Template*

> .. In your local file manager, navigate to the folder where your templates.yaml file is residing and click *Select*.

Reviewer 3: These are appearing as a bulleted list instead of a numbered list in the preview.
```suggestion
.. In your local file manager, navigate to the folder where your `templates.yaml` file is residing and click *Select*.
```

> .. In your local file manager, navigate to the folder where your templates.yaml file is residing and click *Select*.

Reviewer 2: ```suggestion
.. In your local file manager, navigate to the folder where your `templates.yaml` file is stored and click *Select*.
```

> .. On the *Template Editor* page, select the templates.yaml file.

Reviewer 3: ```suggestion
.. On the *Template Editor* page, select the `templates.yaml` file.
```

> .. On the *Template Editor* page, select the templates.yaml file.

Reviewer 2: ```suggestion
.. On the *Template Editor* page, select the `templates.yaml` file.
```

.. In the *Name* field, enter a unique name of the component.

> .. In the *Owner* field, enter the owner of the component and click *Next*.

Reviewer 2: This is fine, but we can probably just say _[On whatever page], enter text into the required fields and click _Next_._  because 1) it is pretty obvious what the user needs to enter into each field since the info isn't very specific; 2) being less specific about each field on the screen helps with maintenance / managing UI changes

> .. In the *Repository Location* tab, in the *Owner* field, specify the organization, repository, or project to which this repository belongs.
> .. In the *Repository Location* tab, in the *Repository field*, enter the name of the repository and click *Review*.

Reviewer 2: ```suggestion
.. On the *Repository Location* tab, in the *Owner* field, specify the organization, repository, or project to which this repository belongs.
.. On the *Repository Location* tab, in the *Repository* field, enter the name of the repository and click *Review*.
```

.. Optional: Modify the YAML definition for the parameters of your template. For more information about these parameters, see <<Creating a Software Template as a YAML file>>.
.. Review the information for accuracy, then click *Create*.
+
[NOTE]
====
* You can use the *File* menu to do any of the following actions:
** *Open template directory*
** *Create template directory*
** *Close template editor*
* Optional: To customize the fields in the template, click the *Custom Fields Explorer* button.
* Optional: To see the Installed Actions documentation, click the *Installed Actions Documentation* button.

> ====

Reviewer 2: Not sure this needs to be an admonition. It seems a tiny bit random here and it's not entirely clear how this relates to the rest of the procedure. Is it an option?  Let's try weaving this into the main text. Depending on what we're trying to communicate, you can probably write something like _Alternatively, you can use the *File* menu..._
